### PR TITLE
Add possibility to ignore args/kwargs in memoize decorator

### PR DESCRIPTION
**Pull request by oldrichsmejkal**
Example call:
   @cache.memoize(ignored_args = [0], ignored_kwargs = ['arg4'])

### CODE REVIEW

**Comment by grantjenks** [comment]
What’s the use case? Please describe or share a compelling example.

**Comment by grantjenks** [comment]
Here’s what a quick Google search turned up:

- https://stackoverflow.com/questions/30730983/make-lru-cache-ignore-some-of-the-function-arguments
- https://www.reddit.com/r/learnpython/comments/3v75g4/using_functoolslru_cache_only_on_some_arguments/

The first example is compelling to me as it’s a problem I’ve had before too.

**Review by grantjenks — CHANGES_REQUESTED** [review]

File: diskcache/core.py [hunk 1]
```diff
@@ -1894,7 +1898,7 @@ def wrapper(*args, **kwargs):
 
             def __cache_key__(*args, **kwargs):
                 "Make key for cache given function arguments."
-                return args_to_key(base, args, kwargs, typed)
+                return args_to_key(base, args, kwargs, typed, ignored_args, ignored_kwargs)
```
[the remark this answers is not in the document]
Comment (oldrichsmejkal) [reply diskcache/core.py]: Should be ready now

**Comment by grantjenks** [comment]
I tried tonight to find a clever way around the `ignore_arguments` parameter using closures. There are some tricks we can play with Python but they're awfully confusing and not nearly as simple as what's here.

This is just to remind myself to merge the PR.

**Comment by grantjenks** [comment]
Superseded by #226

**Pull request closed without merging**